config/duration.go

```diff
@@ -0,0 +1,54 @@
+package config
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/0xERR0R/blocky/log"
+	"github.com/hako/durafmt"
+	"golang.org/x/exp/constraints"
+)
+
+type Duration struct{ time.Duration }
```

Comment:
I like the idea to move Duration to a new file and to provide some convenience methods/functions to make the code more readable 👍 

One think, I don't understand is, why did you change the Duration type to the embedded struct? I played a bit on your branch and changed the duration.go to following:
```go
package config

import (
	"strconv"
	"time"

	"github.com/0xERR0R/blocky/log"
	"github.com/hako/durafmt"
	"golang.org/x/exp/constraints"
)

type Duration time.Duration

func NewDuration[T constraints.Integer | time.Duration](value T) Duration {
	return Duration(time.Duration(value))
}

func (c Duration) Cast() time.Duration {
	return time.Duration(c)
}

func (c Duration) IsZero() bool {
	return c.Cast() == 0
}

func (c Duration) Seconds() float64 {
	return c.Cast().Seconds()
}

func (c Duration) SecondsU32() uint32 {
	return uint32(c.Cast().Seconds())
}

func (c Duration) String() string {
	return durafmt.Parse(c.Cast()).String()
}

func (c *Duration) UnmarshalText(data []byte) error {
	input := string(data)

	if minutes, err := strconv.Atoi(input); err == nil {
		// number without unit: use minutes to ensure back compatibility
		*c = NewDuration(time.Duration(minutes) * time.Minute)

		log.Log().Warnf("Setting a duration without a unit is deprecated. Please use '%s min' instead.", input)

		return nil
	}

	duration, err := time.ParseDuration(input)
	if err == nil {
		*c = NewDuration(duration)

		return nil
	}

	return err
}

```

so basically I reverted the type definition (Duration is alias for time.Duration), leaved your methods and added the "Seconds" function. With this change, we can use your improvements, but without changing the default tag to JSON. Or did I miss something?

Reply:
I had the same question while reading it.

Reply:
Yeah that's a good middle ground!

I didn't do it like that to avoid needing to wrap every method of `time.Duration`, but we don't use that many so this seems better :)

Reply:
Used the code you posted as a starting point for the continuation. Thanks!